### PR TITLE
[DDW-736] Fix PopOver click behavior

### PR DESCRIPTION
**Pull request by renanvalentin**
This PR fixes the popover click behavior on stake pool listing.

## Screenshots

In the current state the user is unable to re-open the popover unless the mouse leaves the thumb pool.

current-bhv

**After fix**
pool-fixed

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1635791306059500)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/staking/widgets/PoolPopOver.js [hunk 1]
```diff
@@ -33,25 +44,36 @@ export function PoolPopOver(props: {
...
-  const [isHovered, setIsHovered] = useState(false);
+  const [state, setState] = useState<State>(initialState);
+
+  useEffect(() => {
+    if (!state.isHovered && state.isClicked) {
+      setState({ isHovered: true, isClicked: false });
+    }
+  }, [state]);
+
+  const close = useDebouncedCallback(() => {
```
Comment (renanvalentin) [inline source/renderer/app/components/staking/widgets/PoolPopOver.js:55]: Prevents `close` method being from invoked twice. e.g. `onHide` followed by `onClickOutside`

**Comment by gabriela-ponce** [comment]
@renanvalentin Overall, the fix looks good. Here's a small issue I found:

- When clicking on "Delegate to this pool" [there is a delay on hiding the popover,](https://drive.google.com/file/d/12oUps36pkN9wxLMYkLn3otegkD8Ej7sv/view?usp=sharing) so it's displayed over the dialog. [Here's how it looks on production.](https://drive.google.com/file/d/1LySqrceE8ej_tucWAyQOX1nzDylBX90Z/view?usp=sharing)

**Comment by miorsufianiohk** [comment]
Hi @renanvalentin , on build 19506, apart from what Gabriela has found, I noticed that in the list view, one could only open and close the popover when the row background colour changes to grey. However, at time when you hover the mouse inside the row, the row background colour doesn't change to grey making opening/closing popover unsuccessful. Please see video [recording here](https://drive.google.com/file/d/11nx1JQ-uBx6FLdtjxuGjdGn9_M6ARvrt/view?usp=sharing) (Please let me know if you require more information, thank you)

**Comment by alexander-rukin** [comment]
image
Probably it's because of this

**Comment by mchappell** [comment]
I think this work is a duplicate of one of the tasks listed in under DDW-680
The change can be seen [here](https://github.com/input-output-hk/daedalus/pull/2685/files#diff-716867b51dbece372088210c1b5eed6db54e833962b3adbf8409082a1a1402f9)

**Comment by miorsufianiohk** [comment]
> I think this work is a duplicate of one of the tasks listed in under DDW-680 The change can be seen [here](https://github.com/input-output-hk/daedalus/pull/2685/files#diff-716867b51dbece372088210c1b5eed6db54e833962b3adbf8409082a1a1402f9)

I agree with @mchappell . The work on this PR has been addressed in [DDW-680](https://github.com/input-output-hk/daedalus/pull/2685/files#diff-716867b51dbece372088210c1b5eed6db54e833962b3adbf8409082a1a1402f9). I had a look at DDW-680 and could confirm that the PopOver issue has been fixed there. 👍

**Comment by dmitrii-gaico** [comment]
@renanvalentin Hello. Just to clear the things up - the only issue which needs your attention and fix is reported by @gabriela-ponce [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775).
Thank you in advance.

**Comment by miorsufianiohk** [comment]
> @renanvalentin Hello. Just to clear the things up - the only issue which needs your attention and fix is reported by @gabriela-ponce [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775). Thank you in advance.

@dmitrii-gaico , I don't think Gabriela's issue exist on DDW-680. I tested on build 19509 and the latest build 19512 but couldn't reproduce. Could you confirm again please just in case I have missed anything, thanks 👍

**Comment by dmitrii-gaico** [comment]
> > @renanvalentin Hello. Just to clear the things up - the only issue which needs your attention and fix is reported by @gabriela-ponce [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775). Thank you in advance.
> 
> @dmitrii-gaico , I don't think Gabriela's issue exist on DDW-680. I tested on build 19509 and the latest build 19512 but couldn't reproduce. Could you confirm again please just in case I have missed anything, thanks 👍

@miorsufianiohk That's correct. Gabriela's [issue](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775) was introduced in the current PR - [DDW-736] Fix PopOver click behavior. For now this is the only issue we need to fix in framework of DDW-736. That's why i asked to ignore all the previous discussions and focus only on Gabriela's issue. 
Am I missing something?
Thanks!

**Comment by miorsufianiohk** [comment]
> > > @renanvalentin Hello. Just to clear the things up - the only issue which needs your attention and fix is reported by @gabriela-ponce [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775). Thank you in advance.
> > 
> > 
> > @dmitrii-gaico , I don't think Gabriela's issue exist on DDW-680. I tested on build 19509 and the latest build 19512 but couldn't reproduce. Could you confirm again please just in case I have missed anything, thanks 👍
> 
> @miorsufianiohk That's correct. Gabriela's [issue](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-956586775) was introduced in the current PR - [DDW-736] Fix PopOver click behavior. For now this is the only issue we need to fix in framework of DDW-736. That's why i asked to ignore all the previous discussions and focus only on Gabriela's issue. Am I missing something? Thanks!

@dmitrii-gaico , yes. The work on this PR has been addressed in DDW-680 as explained by @mchappell [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-957475436). I have also tested the functionality on DDW-680 which I have confirmed has been addressed [here](https://github.com/input-output-hk/daedalus/pull/2730#issuecomment-958767536) 👍

**Comment by renanvalentin** [comment]
Closing this PR as it was already fixed on [PR 2685 - [DDW-680] Small UI/UX fixes](https://github.com/input-output-hk/daedalus/pull/2685 )

**Pull request closed without merging**